src/basis/BasisManager.h

```diff
@@ -0,0 +1,55 @@
+/******************************************************************************
+ * Copyright (c) 2021.                                                        *
```

Comment:
@bikashkanungo is this commit already pulled into the upstream basis PR?

Reply:
@dsambit Yes, all the basis folder files are already pushed into basis branch, so they should be chained into the PR.